cli/src/commands/undo.rs

```diff
@@ -62,6 +62,10 @@ pub struct UndoArgs {
     what: Vec<RevertWhatToRestore>,
 }
 
+fn tx_description(op: &Operation) -> String {
+    format!("undo operation {}", op.id().hex())
+}
+
 pub fn cmd_undo(ui: &mut Ui, command: &CommandHelper, args: &UndoArgs) -> Result<(), CommandError> {
     if args.operation != "@" {
```

Comment:
Shouldn't we do `op revert` (with "revert operation" message) if `args.operation` is specified? I don't think it is an "undo" operation that can form a stack.

Oh, and it's probably safer to do `op revert` when `--what` is specified. We wouldn't want to check if "undo" operations within the stack matches the current `--what` flag.

Reply:
Oh, you're right. I thought that's what I did, but I passed the wrong `tx_description`. I'll add a test about this.

Does that mean we should deprecate `--what` on `jj undo` as well?

Reply:
Alright, I take it that means also deprecate it on `jj op revert`. I will do that.

Reply:
I assume I should do the deprecation of `--what` in a separate PR, let me know if you want it included in this one.

Reply:
Yes, seems better to do that in a separate PR so it doesn't block this one